p2p/5.Create_an_ERC-20/README.md

```diff
@@ -0,0 +1,245 @@
+[PASS] testTransfer() (gas: 36491)
+Test result: ok. 5 passed; 0 failed; finished in 1.72ms
+```
+
+If not, fix the function implementation.
+
+## BONUS Step 5 - Let's deploy it
+
+### 📑 **Description**:
+
+In this step, you will deploy your ERC-20 on Polygon's Mumbai Testnet. This step requires a lot of tools that's why it's a bonus step.
+
+### 📌 **Tasks**:
+
+- Download [Metamask](https://metamask.io) and create an account, if you don't already have one.
+- Create your API key on [Alchemy](https://dashboard.alchemy.com/apps).
+    - Sign in.
+    - Click on `Create new app`.
+    - Enter a name for your API key.
+    - Select `Polygon PoS` chain and `Polygon Mumbai` network.
+    - Click on `Create app`.
+    - Now, click on `API Key` and you can see your API key as well as your RPC URL in `HTTPS`.
+- Add [Mumbai Testnet](https://www.alchemy.com/overviews/mumbai-testnet) network to Metamask, [here](https://www.alchemy.com/overviews/mumbai-testnet#how-to-get-started-using-the-mumbai-testnet) a doc to explain how to do it.
+- Go to [Mumbai faucet](https://mumbaifaucet.com), enter your wallet address and send you MATIC.
+- Copy the `.env` file which is in the `utils` folder to your project and complete the variables except the last one.
+
+</br>
+
+The setup is now finished, let's go to the interesting part.
+
+- Load the environment variables:
+```
+source .env
+```
+
+- Deploy your ERC-20:
+```bash
+forge create src/ERC20.sol:ERC20 --private-key "$PRIVATE_KEY" --rpc-url $RPC_URL --constructor-args 100000 --legacy
+```
+> 💡 You can replace `100000` by the total number of tokens you want
+
+- Copy the address of `Deployed to` and paste it into the `.env` file in the `CONTRACT_ADDRESS` variable.
+
+- Load again the environment variables:
+```
+source .env
+```
+- Check that the contract has been deployed correctly:
+```bash
+cast call $CONTRACT_ADDRESS "totalSupply()" --rpc-url $RPC_URL
+```
+
+Now, you can go in Metamask on Mumbai Testnet, click on `Import tokens` and put the `CONTRACT_ADDRESS`. Congratulation! You have your own token in your metamask wallet. You can now do what you want with your tokens.
+
+</br>
+
+Some examples of possible interactions:
+
+- To transfer tokens to another address:
+```bash
+cast send $CONTRACT_ADDRESS "transfer(address, uint256)" $WALLET 100 --private-key $PRIVATE_KEY --rpc-url $RPC_URL
+```
+- To see the balance of an address:
+```bash
+cast call $CONTRACT_ADDRESS "balanceOf(address)" $WALLET --rpc-url $RPC_URL
+```
+- Approve an address to transfer tokens:
+```bash
+cast send $CONTRACT_ADDRESS "approve(address, uint256)" $WALLET 200 --private-key $PRIVATE_KEY --rpc-url $RPC_URL
+```
+
+### 📚 **Documentation**:
+
+- [Metamask](https://metamask.io)
+- [Mumbai Testnet](https://www.alchemy.com/overviews/mumbai-testnet)
+- [Add Mumbai Testner to Metamask](https://www.alchemy.com/overviews/mumbai-testnet#how-to-get-started-using-the-mumbai-testnet)
+- [Mumbai faucet](https://mumbaifaucet.com/)
+
```

Comment:
as bonuses, you could for example add things like:
- discovering ERC-721 contracts
- deployment to other EVM chains
- ...